homeassistant/components/mazda/__init__.py

```diff
@@ -67,12 +70,15 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         _LOGGER.error("Error occurred during Mazda login request: %s", ex)
         raise ConfigEntryNotReady from ex
 
-    async def async_handle_service_call(service_call=None):
+    async def async_handle_service_call(service_call: ServiceCall) -> None:
         """Handle a service call."""
         # Get device entry from device registry
         dev_reg = device_registry.async_get(hass)
         device_id = service_call.data["device_id"]
         device_entry = dev_reg.async_get(device_id)
+        if TYPE_CHECKING:
+            # For mypy: it has already been checked in validate_mazda_device_id
+            assert device_entry
```

Comment:
I don't like these kinds of concepts.
The fact is, `async_get` may return None; which should just be handled; not silenced.

Reply:
I don't like it either, and I originally planned to raise a `ValueError`.
But then I noticed it was already handled inside the schema validation process.